airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/check/commands.py

```diff
@@ -79,9 +76,13 @@ async def python(pipeline_context: ClickPipelineContext) -> CommandResult:
 async def java(pipeline_context: ClickPipelineContext) -> CommandResult:
     """Format java, groovy, and sql code via spotless."""
     dagger_client = await pipeline_context.get_dagger_client(pipeline_name="Check java formatting")
-    container = format_java_container(dagger_client)
-    check_commands = ["./gradlew spotlessCheck --scan"]
-    return await get_check_command_result(check.commands["java"], check_commands, container)
+    dir_to_format = dagger_client.host().directory(".", include=JAVA_INCLUDE_RULES)
+    async with GradleTaskExecutor(dagger_client, dir_to_format) as executor:
+        try:
+            stdout = await (await executor.run_task("spotlessCheck")).stdout()
+            return CommandResult(check.commands["java"], status=StepStatus.SUCCESS, stdout=stdout)
+        except dagger.ExecError as e:
+            return CommandResult(check.commands["java"], status=StepStatus.FAILURE, stderr=e.stderr, stdout=e.stdout, exc_info=e)
```

Comment:
Is it worth factoring this CommandResult constructor boilerplate? It's basically `get_check_command_result` but without the `run_check` call.

Reply:
Will do